packages/backend-tasks/src/tasks/LocalTaskWorker.ts

```diff
@@ -64,6 +64,7 @@ export class LocalTaskWorker {
         this.logger.info(`Task worker finished: ${this.taskId}`);
       } catch (e) {
         this.logger.warn(`Task worker failed unexpectedly, ${e}`);
+        this.start(settings, options)
```

Comment:
The original intent was that this catch was only meant for actually catastrophic situations and that all internal functions should have their own try/catches to "protect" the surroundings.

Yes an outer loop (instead of calling the entire function again) is better. But at that point it should probably have a sleep timer so that it doesn't enter a too-aggressive retry behavior or so.

Reply:
I think having a recursive however would allow log monitoring on this error to be easier and more obvious

Reply:
Another option is I could pass it a different `settings.initialDelayDuration` of lets say 300 that way it ends up sleeping every time for a short bit.

Reply:
Why though? There's no reason to go through the top parts of the code again, and if that new thing persists again then you've long term changed the settings of the task itself. Nah just run a loop of attempts, then we can emit a metric for failure count or meter, as needed, at a later time

Reply:
How does that look now @freben @Rugvip